flatdata-rs/lib/src/rawdata.rs

```diff
@@ -0,0 +1,35 @@
+/// Exposes blocks of raw data, providing auxiliary functionality like
+/// extracting substrings
+#[derive(Debug, Clone, Copy)]
+pub struct RawData<'a> {
+    data: &'a [u8],
+}
+
+impl<'a> std::ops::Deref for RawData<'a> {
+    type Target = [u8];
+
+    fn deref(&self) -> &[u8] {
+        self.data
+    }
+}
+
+impl<'a> RawData<'a> {
+    /// Create a new object from raw memory reference
+    pub fn new(data: &'a [u8]) -> Self {
+        Self { data }
+    }
+
+    /// Read a \0 terminated substring starting at specified offset
+    pub fn substring(&self, start: usize) -> Result<&str, std::str::Utf8Error> {
```

Comment:
Should we also add `substring_unchecked`?

Reply:
Not for now, we have no unchecked version of any other function (at_unchecked, slice_unchecked, etc). Can however consider adding a bunch of unsafe functions if power-user need those

Reply:
Makes sense.